nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/ControllerResource.java

```diff
@@ -449,6 +455,72 @@ public Response createReportingTask(
+    @ApiOperation(
+            value = "Imports a reporting task snapshot",
+            response = VersionedReportingTaskImportResponseEntity.class,
+            authorizations = {
+                    @Authorization(value = "Write - /controller")
+            }
+    )
+    @ApiResponses(
+            value = {
+                    @ApiResponse(code = 400, message = "NiFi was unable to complete the request because it was invalid. The request should not be retried without modification."),
+                    @ApiResponse(code = 401, message = "Client could not be authenticated."),
+                    @ApiResponse(code = 403, message = "Client is not authorized to make this request."),
+                    @ApiResponse(code = 409, message = "The request was valid but NiFi was not in the appropriate state to process it. Retrying the same request later may be successful.")
+            }
+    )
+    public Response importReportingTaskSnapshot(
+            @Context final HttpServletRequest httpServletRequest,
+            @ApiParam(
+                    value = "The import request containing the reporting task snapshot to import.",
+                    required = true
+            ) final VersionedReportingTaskImportRequestEntity importRequestEntity) {
+
+        if (importRequestEntity == null || importRequestEntity.getReportingTaskSnapshot() == null) {
+            throw new IllegalArgumentException("Reporting task snapshot is required");
+        }
+
+        final VersionedReportingTaskSnapshot requestSnapshot = importRequestEntity.getReportingTaskSnapshot();
+        serviceFacade.discoverCompatibleBundles(requestSnapshot);
+        serviceFacade.generateIdentifiersForImport(requestSnapshot, () -> generateUuid());
+
+        if (isReplicateRequest()) {
+            return replicate(HttpMethod.POST, importRequestEntity);
+        } else if (isDisconnectedFromCluster()) {
+            verifyDisconnectedNodeModification(importRequestEntity.getDisconnectedNodeAcknowledged());
+        }
+
+        return withWriteLock(
+                serviceFacade,
+                importRequestEntity,
+                lookup -> {
+                    authorizeController(RequestAction.WRITE);
+
+                    final Set<ConfigurableComponent> restrictedComponents = FlowRegistryUtils.getRestrictedComponents(requestSnapshot, serviceFacade);
+                    restrictedComponents.forEach(restrictedComponent -> {
+                        final ComponentAuthorizable restrictedComponentAuthorizable = lookup.getConfigurableComponent(restrictedComponent);
+                        authorizeRestrictions(authorizer, restrictedComponentAuthorizable);
+                    });
+                },
+                () -> {
+                    // Nothing to verify
+                },
+                (importRequest) -> {
+                    final VersionedReportingTaskSnapshot snapshot = importRequestEntity.getReportingTaskSnapshot();
```

Comment:
Need to ensure that we're referencing `importRequest` here, not `importRequestEntity`. Otherwise, the request could be changed between first and second phase of the two-phase protocol